### ISSUE TITLE
Crash when I enabled a realistic braking

### ISSUE THREAD

**Issue by telk5093** [JGRennison/OpenTTD-patches#215]
## Version of OpenTTD
0.40.1

## Expected result
No crash

## Actual result
Crash

## Steps to reproduce
I enabled a realistic braking on my savegame and it crashed.
I attached the crash log & emergency save file.

![image](https://user-images.githubusercontent.com/10726524/107510486-5a356c00-6be7-11eb-88d3-6d4974da3dc2.png)
[crash-20210210T123143Z.zip](https://github.com/JGRennison/OpenTTD-patches/files/5958139/crash-20210210T123143Z.zip)

**Comment by James103** [comment]
I'm missing the following NewGRF files when loading the savegame:
```
dbg: [grf] NewGRF 4B520002 (k_bridges v1.2.grf) not found; checksum 7872EA1EA4647D94B8CFF81CE83B393D, name: 'Korea Bridge'
dbg: [grf] NewGRF 52540001 (burj.grf) not found; checksum 7B187296C700FE2BC2C4028369A315C0, name: 'Burj al Arab'
```

**Comment by JGRennison** [comment]
Thanks for reporting this, do you have any more saves from before the time of the crash?
How long ago did you switch on realistic braking on, and did you do any other actions before the crash?

Train 1179 is in a bad state, and it is not clear what sequence of actions resulted in that state.
To get round the crash in this case, you can toggle realistic braking off then on again. You will also need to get rid of its reservation starting from 297x1890.

**Comment by telk5093** [comment]
@James103 Here is a download link: 
 + [k_bridges v1.2.grf](https://telk.kr/ottd/newgrf/archive/download/829/1)
 + [burj.grf](https://telk.kr/ottd/newgrf/archive/download/790/1)

@JGRennison [Telkworld v3.4.7.zip](https://github.com/JGRennison/OpenTTD-patches/files/5967240/Telkworld.v3.4.7.zip) is the original map without using realistic brake. I enabled realistic brake to this savegame and it crashes after time goes.
And thanks for temporary solution :)

**Comment by JGRennison** [comment]
It seems likely that this was fixed by 1b0e602ed5c5c360bb9158baeb22ac37617025f2. If this occurs again in 0.40.2 or later this can be reopened.
